pkg/kv/kvserver/client_raft_test.go

```diff
@@ -6324,3 +6325,531 @@ func TestRaftCheckQuorum(t *testing.T) {
 		})
 	})
 }
+
+// getFirstStoreMetrics retrieves the count of each store metric specified in
+// the metricsName parameter for the first store of the target server and
+// returns the result as a map. The keys in the map correspond to the strings
+// provided in metricsName. The corresponding values indicate the count of each
+// metric.
+func getFirstStoreMetrics(
+	t *testing.T, tc *testcluster.TestCluster, serverIdx int, metricsNames []string,
+) map[string]int64 {
+	metrics := make(map[string]int64)
+	for _, metricName := range metricsNames {
+		metrics[metricName] = getFirstStoreMetric(t, tc.Server(serverIdx), metricName)
+	}
+	return metrics
+}
+
+// getMapsDiff returns the difference between the values of corresponding
+// metrics in two maps. Assumption: beforeMap and afterMap contain the same set
+// of keys.
+func getMapsDiff(beforeMap map[string]int64, afterMap map[string]int64) map[string]int64 {
+	diffMap := make(map[string]int64)
+	for metricName, beforeValue := range beforeMap {
+		if v, ok := afterMap[metricName]; ok {
+			diffMap[metricName] = v - beforeValue
+		}
+	}
+	return diffMap
+}
+
+// TestFilterRaftHandler verifies that FilterRaftHandler correctly discards all
+// incoming and outgoing messages when filterRaftHandlerFuncs is set to nil.
+func TestFilterRaftHandler(t *testing.T) {
```

Comment:
I don't think you need a new test if you use the existing thing. Generally great that you've thought about writing a test here, though!